CHANGES.md

```diff
@@ -17,6 +17,10 @@
   section](https://galoisinc.github.io/cryptol/master/FFI.html#the-abstract-calling-convention)
   for more information.
 
+* Allow an explicit `;` separator between `case` branches. This change removes the
+  unreachable code in the grammar for `case` and `where` expressions with explicit
+  `{` and `}` that was never reachable due to the way the layout rule worked.
```

Comment:
A follow-up question is: do we also want to support the ability to write things like `case x of { ...; ... }` with explicit braces? I am inclined to answer "yes" to this question, given that GHC also supports this, and we are largely stealing GHC's syntactic quirks. (This is not something that we need to do as part of this PR, given that the code path that supported parsing `case x of { ...; ... }` never worked in the first place, but we should open a follow-up issue about this.)